### PR TITLE
drivers: power: lt8722: Add driver support for LT8722

### PR DESCRIPTION
**Pull request by rccalam**
## Pull Request Description

The LT8722 is a high performance, high efficiency, monolithic full bridge DC/DC
converter. One side of the full bridge is driven by a pulse width modulation
(PWM) buck power stage, while the other side of the full bridge is driven by
a linear power stage. The LT8722 can deliver up to ±54W of power to its load
while only requiring a single inductor. An integrated 25-bit digital-to-analog
converter (DAC) is used to control the LT8722 output voltage. Two additional
9-bit DACs control the positive and negative output current limits. An analog
output telemetry pin can be used to monitor SPI selectable parameters such as
VIN, VOUT, IOUT or the LT8722 junction temperature. The serial peripheral
interface (SPI) can be used to configure and control the LT8722 allowing for
flexibility to set the desired output voltage, output current limits, voltage
limits, switching frequency and control ON/OFF behavior. The SPI operates at up
to 10MHz allowing for fast readback and control. The LT8722 operates from a
single 3.1V to 15V supply. Silent Switcher techniques are used to minimize
EMI/EMC emissions while delivering high efficiency at high switching
frequencies. The LT8722 is available in a 3mm × 3mm LQFN package.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [ ] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Comment by rccalam** [comment]
V2:

- Use no_os_put_unaligned_be/no_os_get_unaligned_be functions when applicable.
- Rename data read/write to reg read/write.
- Add null checks to functions.
- Use correct format for multiline comments.
- Use sub unit (nanovolt) for the DAC to voltage conversion function.

**Comment by buha** [comment]
stm32 builds are failing, can you look into that please ?

**Comment by buha** [comment]
I merged another PR which created conflicts with yours, please rebase and solve the conflicts.

Also, the last 5 commits contain changes following pull-request review and so you should squash the changes in the corresponding initial commit. Changes to the device driver should all be squashed and go into the device driver commit. Changes to the iio extension to the device driver should go into the iio extension commit. In total you should have just the 5 commits as you originally intended.

File: drivers/power/lt8722/lt8722.c [hunk 1]
```diff
@@ -0,0 +1,1044 @@
...
+	return (LT8722_DAC_OFFSET - voltage) * (1 << LT8722_DAC_RESOLUTION) /
+	       LT8722_DAC_VREF;
+}
+
+/**
+ * @brief Convert DAC code to voltage.
+ * @param dac - DAC code.
+ * @return Voltage value.
+ */
+double lt8722_dac_to_voltage(int32_t dac)
```
[the remark this answers is not in the document]
Comment (CiprianRegus) [reply drivers/power/lt8722/lt8722.c:102]: Yes, all non-integer intermediary values in the driver should be handled as integers in subunits.

**Comment by rccalam** [comment]
V3:
- Use integer subunits to handle all non-integer values.
- Rebase to latest main branch and fixed conflicts.
- Squash commits.

**Comment by CiprianRegus** [comment]
Please update the commit messages in order to only keep the relevant information after the squash.

**Comment by buha** [comment]
FYI, first commit description says this:

```
 drivers: power: lt8722: Add driver support for LT8722

Add initial header and source files for LT8722 driver.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Use no_os_put/get_unalignled_be built-in functions.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Rename data_read/write to reg_read/write.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Add null checks to functions.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Use correct format for multiline comments.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Return DAC voltage in nanovolts.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>

Use subunits to represent non-integer values.

Signed-off-by: Ramon Cristopher Calam <ramoncristopher.calam@analog.com>
```

Second commit is affected as well

**Comment by rccalam** [comment]
V3:
- Updated the commit messages

**Review by buha — APPROVED** [review]

**Pull request merged**